"Cannot set statistical weights" error running test test_screen19

It looks like it was probably caused by the change in https://github.com/dials/dials/pull/1297

```
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/conda_base/lib/python3.8/concurrent/futures/process.py", line 239, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/conda_base/lib/python3.8/concurrent/futures/process.py", line 198, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/conda_base/lib/python3.8/concurrent/futures/process.py", line 198, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/modules/dials/algorithms/indexing/bravais_settings.py", line 328, in refine_subgroup
    refinery, refined, outliers = refine(
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/modules/dials/algorithms/indexing/refinement.py", line 15, in refine
    refiner = RefinerFactory.from_parameters_data_experiments(
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/modules/dials/algorithms/refinement/refiner.py", line 273, in from_parameters_data_experiments
    return cls._build_components(params, reflections, experiments)
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/modules/dials/algorithms/refinement/refiner.py", line 336, in _build_components
    refman = ReflectionManagerFactory.from_parameters_reflections_experiments(
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/modules/dials/algorithms/refinement/reflection_manager.py", line 323, in from_parameters_reflections_experiments
    return refman(
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/modules/dials/algorithms/refinement/reflection_manager.py", line 437, in __init__
    self._weighting_strategy.calculate_weights(self._reflections)
  File "/scratch/jenkins_agent/workspace/xia2_i19_tests/build_dials/modules/dials/algorithms/refinement/weighting_strategies.py", line 24, in calculate_weights
    raise DialsRefineConfigError(
dials.algorithms.refinement.DialsRefineConfigError: Cannot set statistical weights as some indexed reflections have observed variances equal to zero
"""

The above exception was the direct cause of the following exception:

dials_data = <DataFetcher: /dls/science/groups/scisoft/DIALS/dials_data>
tmpdir = local('/scratch/jenkins_agent/workspace/xia2_i19_tests/tests/pytest/test_screen190')

    def test_screen19(dials_data, tmpdir):
        """An integration test.  Check the full functionality of screen19."""
        data_dir = dials_data("x4wide").join("X4_wide_M1S4_2_####.cbf:1:30").strpath
    
        # Test screen19 first.
        with tmpdir.as_cwd():
>           Screen19().run([data_dir], set_up_logging=True)

../screen19/tests/test_screen19.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../screen19/screen19/screen.py:1160: in run
    self._refine_bravais()
../screen19/screen19/screen.py:985: in _refine_bravais
    refined_settings = refined_settings_from_refined_triclinic(
../build_dials/modules/dials/algorithms/indexing/bravais_settings.py:244: in refined_settings_from_refined_triclinic
    for i, result in enumerate(
../build_dials/conda_base/lib/python3.8/concurrent/futures/process.py:484: in _chain_from_iterable_of_lists
    for element in iterable:
../build_dials/conda_base/lib/python3.8/concurrent/futures/_base.py:611: in result_iterator
    yield fs.pop().result()
../build_dials/conda_base/lib/python3.8/concurrent/futures/_base.py:439: in result
    return self.__get_result()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Future at 0x7f3908d3c100 state=finished raised DialsRefineConfigError>

    def __get_result(self):
        if self._exception:
>           raise self._exception
E           dials.algorithms.refinement.DialsRefineConfigError: Cannot set statistical weights as some indexed reflections have observed variances equal to zero

../build_dials/conda_base/lib/python3.8/concurrent/futures/_base.py:388: DialsRefineConfigError
```

--- a/newsfragments/53.bugfix
+++ b/newsfragments/53.bugfix
@@ -1,0 +1,2 @@
+Fix a case of screen19 failing when trying to refine the different possible Bravais settings of the found lattice.
+This was happening because screen19 was failing to filter out reflections with zero variance in the position of their observed centroids before performing the refinement.

--- a/screen19/screen.py
+++ b/screen19/screen.py
@@ -981,9 +981,18 @@ class Screen19(object):
             self.refls = eliminate_sys_absent(self.expts, self.refls)
             map_to_primitive(self.expts, self.refls)
 
+            # We can only refine against reflections with non-zero variance in
+            # observed centroid position.
+            x_valid, y_valid, z_valid = [
+                part > 0 for part in self.refls["xyzobs.mm.variance"].parts()
+            ]
+            nonzero_variance = x_valid | y_valid | z_valid
+
             try:
                 refined_settings = refined_settings_from_refined_triclinic(
-                    self.expts, self.refls, self.params.dials_refine_bravais
+                    self.expts,
+                    self.refls.select(nonzero_variance),
+                    self.params.dials_refine_bravais,
                 )
             except RuntimeError as e:
                 warning("dials.refine_bravais_settings failed.\nGiving up.")

